commands/fun/bye.js

```diff
@@ -18,14 +19,41 @@ module.exports = class extends Command {
       msg.react("🖕")
       return
     }
-    const checkPeopleAreInChannel = []
-    msg.member.voice.channel.members.map(member => {
-      checkPeopleAreInChannel.push(member.user.id)
-    })
+
+    if (!msg.channel.permissionsFor(msg.guild.roles.everyone).has("VIEW_CHANNEL")) {
+      msg.channel.send("To use this command it needs to be posted in a channel that EVERYONE can see. If your doing this in a secret channel you know who you are, naughty naughty")
+      return
+    }
+
+    if (!(msg.channel.name === "general")) {
+      msg.channel.send("Post this in general please, you sneaky beaver")
+      return
+    }
+
+    const byeDbStatus = await this.client.db.getByeStatus()
```

Comment:
You can store the status of the bye against the server object: `msg.guild.bye`, that way if the server dies for whatever reason so does the state.